app/code/Magento/Catalog/Model/Indexer/Product/Price/AbstractAction.php

```diff
@@ -405,7 +405,7 @@ protected function _reindexRows($changedIds = [])
                     $this->tableMaintainer->createMainTmpTable($dimensions);
                     $temporaryTable = $this->tableMaintainer->getMainTmpTable($dimensions);
                     $this->_emptyTable($temporaryTable);
-                    $indexer->executeByDimensions($dimensions, \SplFixedArray::fromArray($entityIds, false));
+                    $indexer->executeByDimensions($dimensions, new \ArrayIterator($entityIds));
```

Comment:
could we use something like 
`                    $entityIds = method_exists(\SplFixedArray::class, 'getIterator') ?
                        \SplFixedArray::fromArray($entityIds, false)->getIterator():
                        \SplFixedArray::fromArray($entityIds, false);`

Reply:
@xmav if I’m not mistaken, fromArray has been removed from the mentioned class. However ‘ArrayIterator’ implements the same interface and I personally don’t see any issues using this class as an alternative.